File: packages/urdu-js/src/sorting/sort.ts
import { sortKey } from './collation.js'

export function compare(a: string, b: string): number {
  const ka = sortKey(a)
  const kb = sortKey(b)
  return ka < kb ? -1 : ka > kb ? 1 : 0
}

export function sort(words: string[]): string[] {
  return [...words].sort(compare)
}

File: packages/urdu-dart/lib/src/sorting/sort.dart
import 'collation.dart';

/// Compares two Urdu strings lexicographically.
int compare(String a, String b) {
  final ka = sortKey(a);
  final kb = sortKey(b);
  return ka.compareTo(kb);
}

/// Sorts a list of Urdu strings lexicographically.
List<String> sort(List<String> words) {
  final copy = List<String>.from(words);
  copy.sort(compare);
  return copy;
}
